core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -77,6 +80,14 @@ public static MetricsConfig getDefault() {
   }
 
   public static MetricsConfig fromProperties(Map<String, String> props) {
```

Comment:
I've been thinking about this change while doing recent work around writer factories and deletes.

Naming-wise, I'd probably prefer calling it `forTable(table)`. If we decide to store [metrics for rows in position deletes](https://github.com/apache/iceberg/pull/2942#discussion_r683715471), we then can provide something like `forPositionDelete(table)` to promote the path and pos columns and apply the regular config for all other columns in optional `row`.

Reply:
Renamed to 'forTable'